hs_core/templatetags/hydroshare_tags.py

```diff
@@ -90,7 +105,7 @@ def resource_first_author(content):
     elif first_creator.name:
         return format_html('<span>{name}</span>', name=first_creator.name)
     else:
-        first_creator = content.metadata.creators.filter(order=1).first()
+        first_creator = content.content_object.creators.filter(order=1).first()
```

Comment:
Why this change? The query remains the same.

The problem of the extra query is in line 100. We need to replace that line with something like this:
```
first_creator = None
for cr in content.metadata.creators.all(): 
      if cr.order == 1:
            first_creator = cr
            break
```

Reply:
I thought that `content.metadata` would result in an extra query since we don't already explicitly fetch it?

I thought that using the `content_object` would be safe because we already explicitly fetch that
https://github.com/hydroshare/hydroshare/blob/2368-my-res-no-annotate/hs_core/views/utils.py#L670

Reply:
`metadata` is a property (function) of the Resource class that access the `content_object` attribute of the Resource class. So `res.metadata` is same as `res.content_object` 

The problem with `content.first_creator` is that it runs a `filter()` on the existing queryset (`res.metadata.creators`) that results in a new query.  So instead of using  `content.first_creator`, we can loop through the creators and select the first creator in this template filter.

Reply:
Super helpful @pkdash I get it now, thanks! I will push a commit as you suggest

Reply:
Okay I'm beating a dead horse here. But one more question on this.
Taking a look at [when querysets are evaluated](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#when-querysets-are-evaluated) isn't it true that the `.filter()` here that is run by `.first_creator` called here:
https://github.com/hydroshare/hydroshare/blob/508ae7dc7b617a81d55e0465e3e6d52b7e063e69/hs_core/templatetags/hydroshare_tags.py#L100
**_wouln't actually hit the database again_**? 

However, [iteration does hit the db](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#when-querysets-are-evaluated) so wouldn't the change you suggested above 
>for cr in content.metadata.creators.all():
...

result in an additional db hit?

Reply:
nevermind, I think you're right

Reply:
It seems iterating over prefetched objects doesn't hit the db.

Reply:
100% correct, that's what I found too